src/rank_llm/retrieve/retriever.py

```diff
@@ -216,10 +217,21 @@ def retrieve(
             candidates_file = Path(
                 f"{retrieve_results_dirname}/{self._retrieval_method.name}/retrieve_results_{self._dataset}_top{k}.json"
             )
+
             if not candidates_file.is_file():
-                print(f"Retrieving with dataset {self._dataset}")
-                pyserini = PyseriniRetriever(self._dataset, self._retrieval_method)
-                retrieved_results = pyserini.retrieve_and_store(k=k)
+                try:
+                    file_path = download_encoded_queries(
+                        f"{self._retrieval_method.name}/retrieve_results_{self._dataset}_top{k}.json"
+                    )
+                    with open(file_path, "r") as f:
+                        loaded_results = json.load(f)
+                    retrieved_results = [
+                        Result(r["query"], r["hits"]) for r in loaded_results
+                    ]

```

Comment:
I think the first possibility is reusing the cache, not downloading. If the file is already downloaded and the md5 matches the expectation, we should not download it.